### ISSUE TITLE
Error while running keras(tensorflow backend) model on android 

### ISSUE THREAD

**Issue by Joel-kiran** [Joel-kiran/Digit-Recognition#1]
Iam a beginner and trying to run a keras(tensorflow backend)  model on android, so i have written the following lines of code in the android studio, i am getting these errors when i run the  app.

    E/native: tensorflow_inference_jni.cc:233 Error during inference: Internal: Output 0 of type double does not match declared output type float for node _recv_ip_input_0 = _Recv[client_terminated=true, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=-7543542456069897885, tensor_name="ip_input", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()
    E/native: tensorflow_inference_jni.cc:170 Output [op/Sigmoid] not found, aborting!

I have downloaded the .so and .jar file from "jenkins(https://ci.tensorflow.org/view/Nightly/job/nightly-android/)" with latest build.

kindly review my code to solve the issue:
   

        private static final String MODEL_FILE = "file:///android_asset/opt_dia.pb";
        private static final String INPUT_NODE = "ip_input";
        private static final String OUTPUT_NODE = "op/Sigmoid";
        private static final int[] INPUT_SIZE = {1,8};
        private TensorFlowInferenceInterface inferenceInterface;
    
        static {System.loadLibrary("tensorflow_inference");}
        protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        inferenceInterface = new TensorFlowInferenceInterface();
        inferenceInterface.initializeTensorFlow(getAssets(), MODEL_FILE);
        final Button button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {public void onClick(View v){
        double[] inputFloats={1,85,34,29,0,26.6,0.351,50};
        inferenceInterface.fillNodeDouble(INPUT_NODE, INPUT_SIZE, inputFloats);
        inferenceInterface.runInference(new String[] {OUTPUT_NODE});
        double[] resu ={2};//This node ideally holds the classification value after performing inferenceInterface.runInference
        inferenceInterface.readNodeDouble(OUTPUT_NODE,resu);//To get the classification result(0 or 1) by reading the output node value from model.
        })};
// so when I click the run button on the app, I get the above two errors, it’s not updating the resu(output) array.
So basically the value is not read, when this operation (inferenceInterface.readNodeDouble(OUTPUT_NODE,resu))is performed,the resu array is not updated, it gives the above error when this line is parsed and the output is always 2 irrespective of the input to the model.
